packages/@uppy/core/src/Uppy.js

```diff
@@ -544,34 +590,24 @@ class Uppy {
   addFile (file) {
     this.#assertNewUploadAllowed(file)
 
-    const { files } = this.getState()
-    let newFile = this.#checkAndCreateFileStateObject(files, file)
+    const { newFilesState, validFilesToAdd, errors } = this.#checkAndCreateFileStateObjects([file])
 
-    // Users are asked to re-select recovered files without data,
-    // and to keep the progress, meta and everthing else, we only replace said data
-    if (files[newFile.id] && files[newFile.id].isGhost) {
-      newFile = {
-        ...files[newFile.id],
-        data: file.data,
-        isGhost: false,
-      }
-      this.log(`Replaced the blob in the restored ghost file: ${newFile.name}, ${newFile.id}`)
-    }
+    const restrictionErrors = errors.filter((error) => error.isRestriction)
+    this.#informAndEmit(restrictionErrors)
 
-    this.setState({
-      files: {
-        ...files,
-        [newFile.id]: newFile,
-      },
-    })
+    if (errors.length > 0) throw errors[0]
+
+    const [validFileToAdd] = validFilesToAdd
+
+    this.setState({ files: newFilesState })
 
-    this.emit('file-added', newFile)
-    this.emit('files-added', [newFile])
-    this.log(`Added file: ${newFile.name}, ${newFile.id}, mime type: ${newFile.type}`)
+    this.emit('file-added', validFileToAdd)
+    this.emit('files-added', validFilesToAdd)
```

Comment:
same here, one letter difference is pretty hard to follow. Especially since I'm dyslexic.

Reply:
hmm we can do `validFilesToAdd[0]` but then we have to repeat that many times

Reply:
`validFile` and `validFileCollection`?

Reply:
i changed to validFilesToAdd and firstValidFileToAdd (which is more correct)